processor/src/test/java/org/mapstruct/ap/test/selection/qualifier/QualifierTest.java

```diff
@@ -189,7 +189,12 @@ public void testFactorySelectionWithQualifier() {
                         kind = Kind.ERROR,
                         line = 37,
                         messageRegExp = "'nullValueMappingStrategy', 'resultType' and 'qualifiedBy' are undefined in "
-                            + "@BeanMapping, define at least one of them." )
+                            + "@BeanMapping, define at least one of them."),
+                @Diagnostic(type = ErroneousMovieFactoryMapper.class,
+                    kind = Kind.ERROR,
+                    line = 37,
+                    messageRegExp = "The result type .*\\.AbstractEntry is abstract. Provide non abstract result type" +
+                        " or a factory method.")
```

Comment:
Could you reformat the complete `@ExpectedCompilationOutcome(..)`, please? The formatting is a bit strange above...